### PR TITLE
[FLOSS H3] jna: extend HelloElektra #4093

### PR DESCRIPTION
**Pull request by JakobWonisch**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Review by muskater — COMMENTED** [review]
Nice work, you wrote a comprehensive and easy to understand example. I think this is a nice addition to the Java-Language bindings. My only suggestions would be that you link the `issue` to this PR so it is easier to understand what your are trying to archive. You might also change the title of your PR into something a little bit more descriptive and add a small description in your main comment. The restyling issue should be easily fixable with a reformatting tool. All in all, you did a great job!

**Comment by lawli3t** [comment]
Good PR, there seem to be some styling issues.

One thing I noticed apart from the suggestions from @4ydan is that maybe some comments in the Hello Elektra would be helpful? Not sure if it is necessary, because the functions itself should be explained in the core API documentation, but it would be a nice addition I think.

File: src/bindings/jna/hello/src/main/java/HelloElektra.java [hunk 1]
```diff
@@ -45,5 +45,15 @@ public static void main(String[] args) {
 
     Key n = Key.create("user:/weird\\/name///\\\\/is/\no/_\\\\problem");
     n.keyNameIterator().forEachRemaining(s -> System.out.println("itername: " + s));
+
+    Key cutpoint = Key.create("user:/cutpoint"),
+    	ka = Key.create("user:/cutpoint/hello", "hiback"),
```
Comment (markus2330) [inline src/bindings/jna/hello/src/main/java/HelloElektra.java:50]: As build server says, there is a formatting problem.

File: src/bindings/jna/hello/src/main/java/HelloElektra.java [hunk 2]
```diff
@@ -45,5 +45,15 @@ public static void main(String[] args) {
...
+     	kb = Key.create("user:/cutpoint/hello2", "hellotoo"),
+     	kc = Key.create("user:/different/hello", "hellothere");
+	KeySet whole = KeySet.create(ka, kb, kc),
+		cut = whole.cut(cutpoint);
+
+	ka.setString("hibackagain");
```
Comment (markus2330) [inline src/bindings/jna/hello/src/main/java/HelloElektra.java:56]: Unclear that this is now a different example.

File: src/bindings/jna/hello/src/main/java/HelloElektra.java [hunk 3]
```diff
@@ -45,5 +45,15 @@ public static void main(String[] args) {
 
     Key n = Key.create("user:/weird\\/name///\\\\/is/\no/_\\\\problem");
     n.keyNameIterator().forEachRemaining(s -> System.out.println("itername: " + s));
+
+    Key cutpoint = Key.create("user:/cutpoint"),
+    	ka = Key.create("user:/cutpoint/hello", "hiback"),
+     	kb = Key.create("user:/cutpoint/hello2", "hellotoo"),
+     	kc = Key.create("user:/different/hello", "hellothere");
```
Comment (markus2330) [inline src/bindings/jna/hello/src/main/java/HelloElektra.java:52]: Make more clear what this example shows:

```suggestion
     	kc = Key.create("user:/outside/cutpoint/hello", "hellothere");
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Examples are very minimal and it is not so clear what they demonstrate.

**Comment by markus2330** [comment]
@lawli3t wrote:

> One thing I noticed apart from the suggestions from @4ydan is that maybe some comments in the Hello Elektra would be helpful? 

Do you mean @JakobWonisch here?

**Comment by markus2330** [comment]
Please fix styling issues.

**Pull request merged**